STAR-Delegation/CSR-template/template-schema.cddl

```diff
@@ -0,0 +1,76 @@
+csr-template-schema = {
+  ? keyTypes: [* rsaKeyType / ecKeyType] ; XXX(tho) could IdO specify multiple keyTypes?
+  ? subject: distinguishedName
+  ? extensions: extensions
+}
+
+mandatory-wildcard = "**"
+optional-wildcard = "*"
+wildcard = mandatory-wildcard / optional-wildcard
+
+; regtext matches all text strings but "*" and "**"
+regtext = text .regexp "([^\*].*)|([\*][^\*].*)|([\*][\*].+)"
+
+regtext-or-wildcard = regtext / wildcard
+
+distinguishedName = {
+  ? country: regtext-or-wildcard
+  ? stateOrProvince: regtext-or-wildcard
+  ? locality: regtext-or-wildcard
+  ? organization: regtext-or-wildcard
+  ? organizationalUnit: regtext-or-wildcard
+  ? emailAddress: regtext-or-wildcard
+  ? commonName: regtext-or-wildcard
+}
+
+rsaKeyType = {
+  ? PublicKeyType: "RSA"
+  ? PublicKeyLength: $rsaKeySize / mandatory-wildcard
+  ? SignatureType: $signatureType / mandatory-wildcard
+}
+
+$rsaKeySize /= "2048"
+$rsaKeySize /= "4096"
+$rsaKeySize /= "8192"
+
+$signatureType /= "sha256WithRSAEncryption"
+
+ecKeyType = {
+  ? PublicKeyType: "ecPublicKey"
+  ? Curve: $ecCurve
+  ? SignatureType: $ecSignatureType
+}
+
+$ecCurve /= "secp521r1"
+
+$ecSignatureType /= "ecdsa-with-SHA256"
+
+subjectaltname = {
+  ? DNS: [ 1* regtext ]
```

Comment:
Why not allow a wildcard here? In fact we need to look again at the JSON Schema doc to make sure wildcards are possible in these fields.

Reply:
The thing is that we don't we don't have a way to template the delegated identifier, and since the IdO doesn't want NDC to pick a random name, it *has* to specify it.

Reply:
No, we don't support (and should not support) name templates. But I think we should actually allow the NDC to pick the name, in situations where the IdO trusts it enough. I think these situations will be rare.

Reply:
OK, I appreciate the practical aspect of this choice.  However, this makes the template not self-sufficient for the purpose of checking the contents of the CSR.  I think that the need to check the SAN with some "external" logics needs to be raised in the security considerations.

Reply:
Yes, added to the issue.